src/main/java/org/isf/menu/manager/UserSettingManager.java

```diff
@@ -58,25 +64,50 @@ public UserSetting newUserSetting(UserSetting userSetting) throws OHServiceExcep
 	public UserSetting updateUserSetting(UserSetting userSetting) throws OHServiceException {
 		return userSettingIoOperationRepository.save(userSetting);
 	}
-
+	
 	/**
 	 * Returns the list of {@link UserSetting}s of the specified userId.
 	 *
-	 * @param userId - the user Id
+	 * @param userName - the user name
 	 * @return the list of {@link UserSetting}s
 	 */
-	public List<UserSetting> getUserSetting(String userId) throws OHServiceException {
-		return userSettingIoOperationRepository.findAllByUSerID(userId);
+	public List<UserSetting> getUserSettingByUserName(String userName) throws OHServiceException {
+		return userSettingIoOperationRepository.findAllByUSerName(userName);
 	}
 
 	/**
 	 * Returns {@link UserSetting}s of the specified userId.
 	 *
-	 * @param userId     - the user Id
+	 * @param userName     - the user name
 	 * @param configName - the name of the user setting
 	 * @return <code>{@link UserSetting}</code> if the userSetting exist, <code>null</code> otherwise.
 	 */
-	public UserSetting getUserSetting(String userId, String configName) throws OHServiceException {
-		return userSettingIoOperationRepository.findUserSettingByUSerID(userId, configName);
+	public UserSetting getUserSettingByUserNameConfigName(String userName, String configName) throws OHServiceException {
+		return userSettingIoOperationRepository.findByUserNameAndConfigName(userName, configName);
+	}
+	
+	/**
+	 * Returns {@link UserSetting}s of the specified userId.
+	 *
+	 * @param userName     - the user name
+	 * @return <code>{@link UserSetting}</code> if the userSetting exist, <code>null</code> otherwise.
+	 */
+	public UserSetting getUserSettingById(int userSettingId) throws OHServiceException {
+		Optional<UserSetting> us = userSettingIoOperationRepository.findById(userSettingId);
```

Comment:
why use `Optional` and return `null`; the caller can test using ~ifPresent()` just as easily as testing for `null`